server/src/test/java/org/elasticsearch/action/admin/indices/alias/AliasActionsTests.java

```diff
@@ -311,6 +325,19 @@ public void testRoundTrip() throws IOException {
         }
     }
 
+    public void testFromToXContent() throws IOException {
+        for (int runs = 0; runs < 20; runs++) {
+            AliasActions action = randomAliasAction();
+            XContentType xContentType = randomFrom(XContentType.values());
+            BytesReference shuffled = toShuffledXContent(action, xContentType, ToXContent.EMPTY_PARAMS, false, "filter");
+            AliasActions parsedAction;
+            try (XContentParser parser = createParser(xContentType.xContent(), shuffled)) {
+                parsedAction = AliasActions.fromXContent(parser);
+                assertNull(parser.nextToken());
+            }
+            assertThat(parsedAction, equalTo(action));
+        }
+    }
 
     private Map<String, Object> randomMap(int maxDepth) {
```

Comment:
I see some duplication her between the newly added generator and this method. Maybe we can address that too?